src/Tribe/Aggregator/Record/Abstract.php

```diff
@@ -887,17 +887,26 @@ public function set_status( $status ) {
 			return false;
 		}
 
+		// Prevent to change the status of a schedule import.
+		if ( $this->post instanceof WP_Post && $this->post->post_status === Tribe__Events__Aggregator__Records::$status->schedule ) {
+			return false;
+		}
 
 		$status = wp_update_post( [
 			'ID'          => $this->id,
 			'post_status' => Tribe__Events__Aggregator__Records::$status->{$status},
 		] );
 
-		if ( ! is_wp_error( $status ) && ! empty( $this->post->post_parent ) ) {
-			$status = wp_update_post( [
-				'ID'            => $this->post->post_parent,
-				'post_modified' => date( Tribe__Date_Utils::DBDATETIMEFORMAT, current_time( 'timestamp' ) ),
-			] );
+		if ( ! is_wp_error( $status ) ) {
+			// Reload the properties of the post if the status of the record was changed.
+			$this->load( $this->id );
+
+			if ( ! empty( $this->post->post_parent ) ) {
+				$status = wp_update_post( [
+					'ID'            => $this->post->post_parent,
+					'post_modified' => date( Tribe__Date_Utils::DBDATETIMEFORMAT, current_time( 'timestamp' ) ),
```

Comment:
* :no_entry_sign: date() is affected by runtime timezone changes which can cause date/time to be incorrectly displayed. Use gmdate() instead.
* :warning: Calling current_time() with a $type of "timestamp" or "U" is strongly discouraged as it will not return a Unix (UTC) timestamp. Please consider using a non-timestamp format or otherwise refactoring this code.

<sub>hash: 1d7fb91e9270ec60d19dc422e9b96912</sub>

Reply:
```php
use Tribe__Date_Utils as Dates;
```

```suggestion
					'post_modified' => Dates::build_date_object()->format( Dates::DATETIMEFORMAT ),
```